setup.py

```diff
@@ -54,21 +124,35 @@ def set_builtin(name, value):
 
 # Custom builder to handler compiler flags. Edit if needed.
 class build_ext_subclass(build_ext):
+
+    def _omp_compile_link_args(self, compiler):
+        # Get OpenMP setting from environment
+        use_omp = OMP_SETTING_TABLE[os.environ.get('USE_OMP', 'probe')]
+        # OpenMP is not supported with default clang
+        # Conda provides its own compiler which does support openmp
+        if use_omp in {'probe', 'omp'} and is_macOS() and not is_conda_interpreter():
```

Comment:
As I understand it, gomp is closer to an intrinsic library for GCC while it's an optional extension for Clang. So finding include and link paths is not needed for GCC (e.g. my test case of `CC=gcc-mp-12 USE_OMP=gcc pip install .` uses libgomp as is expected) 

With omp, however, it may wind up in the compiler default search path (Anaconda?) or you may have to be told or go looking for it. Homebrew and macports just provide tools for finding often-used install points, but you can always build it yourself and then set CFLAGS and LDFLAGS. 

I could see an argument, therefore, for only looking for it in the 'probe' case, in. case someone has a specific omp they've built themselves and are specifying those flag variables (i.e. don't probe for USE_OMP=clang case). I'm not sure what the final result would be if you had a homebrew installed libomp, and a manually installed libomp, in other words.

Reply:
Yeah I suppose in that case we should be extremely careful about accidentally pulling in includes/libs from unintended sources.